### PR TITLE
Fixed Product model Attributes-related methods

### PR DESCRIPTION
**Pull request by David-Guillot**
Fix suggested by @michalmarcinkowski in #7196 ; thanks @Pitoune for noticing the bug, and @michalmarcinkowski for helping find a pragmatic solution ;)

| Q               | A
| --------------- | ---
| Bug fix?        | yes |
| New feature?    | no |
| BC breaks?      | no |
| Related tickets | fixes #7196 and #7189  |
| License         | MIT |

### CODE REVIEW

File: src/Sylius/Component/Product/Model/Product.php [hunk 1]
```diff
@@ -247,6 +247,10 @@ public function getAttributes()
      */
     public function addAttribute(AttributeValueInterface $attribute)
     {
+        if (false === $attribute instanceof ProductAttributeValueInterface) {
```
Comment (lchrusciel) [inline src/Sylius/Component/Product/Model/Product.php]: `Assert::isInstanceOf()`

File: src/Sylius/Component/Product/Model/Product.php [hunk 2]
```diff
@@ -258,6 +262,10 @@ public function addAttribute(AttributeValueInterface $attribute)
      */
     public function removeAttribute(AttributeValueInterface $attribute)
     {
+        if (false === $attribute instanceof ProductAttributeValueInterface) {
```
Comment (lchrusciel) [inline src/Sylius/Component/Product/Model/Product.php]: `Assert::isInstanceOf()`

**Comment by David-Guillot** [comment]
Ah sorry, I didn't notice that webmozart/assert was included ;)

**Comment by David-Guillot** [comment]
PR rebased from upstream

File: src/Sylius/Component/Product/Model/Product.php [hunk 3]
```diff
@@ -247,6 +248,12 @@ public function getAttributes()
      */
     public function addAttribute(AttributeValueInterface $attribute)
     {
+        Assert::isInstanceOf(
+            $attribute,
+            'Sylius\Component\Product\Model\ProductAttributeValueInterface',
```
Comment (pjedrzejewski) [inline src/Sylius/Component/Product/Model/Product.php]: We should use ``::class`` here and below, wdyt? :)

Comment (David-Guillot) [inline src/Sylius/Component/Product/Model/Product.php]: I thought of it, but I was thinking it was PHP7 only ; I was wrong, I'll fix that in a minute ;)

**Comment by David-Guillot** [comment]
@pjedrzejewski just a reminder because I don't know if you saw that the changes you requested have been made 4 days ago ;)

**Comment by michalmarcinkowski** [comment]
Thanks David! 👍

**Pull request merged**